New How to run AM3 docs (#1238)
ACCESS-Hive/access-hive.github.io: docs/models/run_a_model/run_access-am3.md

All {{model}} configurations are licensed under the UKMO's Momentum licence. {{model}} is delivered to the community through private GitHub repositories. See the [Request access section](#request-access) for details.

{{ model }} release notes are [available on the ACCESS-Hive Forum]({{release_notes}}) and are updated when new releases are made available.

## Prerequisites

- **NCI account**<br>
    Before running an ACCESS model, you need to [Set Up your NCI Account](/getting_started/set_up_nci_account).

- **Request access to the configurations**<br>
    To request access to the configurations repository (and associated model component repositories), please contact us [through the ACCESS-Hive Forum](https://forum.access-hive.org.au/t/request-access-to-am3-configurations/5580/13). This step will also allow us to ensure you are properly licensed to use the software.
    {: #request-access }

- **Join NCI projects**<br>

    Join the following projects by requesting membership on their respective NCI project pages:

    - [access](https://my.nci.org.au/mancini/project/access/join)
    - [vk83](https://my.nci.org.au/mancini/project/vk83/join)

>     - [xp65](https://my.nci.org.au/mancini/project/xp65/join)

Reviewer: Add project names for clarity.

```suggestion
- [access](https://my.nci.org.au/mancini/project/access/join): ACCESS software sharing
- [vk83](https://my.nci.org.au/mancini/project/vk83/join): ACCESS Models
- [xp65](https://my.nci.org.au/mancini/project/xp65/join): ACCESS Analysis Environments
```

{% set model = "ACCESS-AM3" %}
{% set github_configs = "https://github.com/ACCESS-NRI/access-am3-configs" %}
{% set github_ssh = "git@github.com:ACCESS-NRI/access-am3-configs.git" %}
{% set configs_docs = "https://access-am3-configs.access-hive.org.au" %}
{% set config_branch = "release-n96e" %}
{% set experiment_name = "my-am3-expt" %}
{% set release_notes = "https://forum.access-hive.org.au/t/access-am3-release-information/5446" %}

[PBS job]: https://opus.nci.org.au/display/Help/4.+PBS+Jobs

!!! release

>     This is a [**Beta Release**](/about/releases), intended for use by experienced users and collaborators. Any model configuration and related source code information on this page may change during the release process.

Reviewer 2: ```suggestion
    This is a [**Beta Release**](/about/releases) intended for experienced users and collaborators. Model configurations and related source code described on this page may change during the release process.
```

<div class="text-card-group" markdown>
[:fontawesome-brands-github:{: class="twemoji icon-before-text"} {{ model }} configurations]({{github_configs}}){: class="text-card"}
[![Hive](/assets/ACCESS_icon_HIVE.png){: class="icon-before-text"} {{ model }} configs docs]({{configs_docs}}){: class="text-card"}
[:notepad_spiral:{: class="twemoji icon-before-text"} Release notes]({{release_notes}}){: class="text-card"}
</div>

# Run {{ model }}

## About

The instructions below outline how to run {{ model }} using ACCESS-NRI's software deployment pipeline, specifically designed to run on the [National Computational Infrastructure (NCI)](https://nci.org.au/about-us/who-we-are) supercomputer [_Gadi_](https://nci.org.au/our-systems/hpc-systems).

> If you are unsure whether {{ model }} is the right choice for your experiment, take a look at the overview of [ACCESS Models](/models).

Reviewer 2: ```suggestion
If you are unsure whether {{ model }} is the right choice for your experiment, see the overview of [ACCESS Models](/models).
```

> All {{model}} configurations are licensed under the UKMO's Momentum licence. {{model}} is delivered to the community through private GitHub repositories. See the [Request access section](#request-access) for details.

Reviewer 2: ```suggestion
{{model}} configurations are licensed under the UKMO's Momentum licence and are distributed to the community through private _GitHub_ repositories. See the [Request access section](#request-access) for details.
```

> {{ model }} release notes are [available on the ACCESS-Hive Forum]({{release_notes}}) and are updated when new releases are made available.

Reviewer 2: ```suggestion
{{ model }} release notes are [available on the ACCESS-Hive Forum]({{release_notes}}) and are updated with each new release.
```

## Prerequisites

- **NCI account**<br>
    Before running an ACCESS model, you need to [Set Up your NCI Account](/getting_started/set_up_nci_account).

- **GitHub account**<br>
    Open an account on [GitHub](https://github.com/signup) if you do not have one.

- **Request access to the configurations**<br>

>     To request access to the configurations repository (and associated model component repositories), please contact us [through the ACCESS-Hive Forum](https://forum.access-hive.org.au/t/request-access-to-am3-configurations/5580). This step will also allow guide you towards getting licensed to use the model as required. 

Reviewer 2: ```suggestion
    To request access to the configurations repository and associated model component repositories, contact us through [this post on the ACCESS-Hive Forum](https://forum.access-hive.org.au/t/request-access-to-am3-configurations/5580). We can also guide you through the licensing process required to use the model. 
```

    {: #request-access }

    !!! warning

        This process can take up to 2 weeks.

- **Join NCI projects**<br>

>     Join the following projects by requesting membership on their respective NCI project pages:

Reviewer 2: ```suggestion
    Request membership to the following NCI projects through their respective project pages:
```

       start="<!--start:cylc8-prerequisites-->"
       end="<!--end:cylc8-prerequisites-->"
    %}

    For more information on joining specific NCI projects, refer to [How to connect to a project](https://opus.nci.org.au/display/Help/How+to+connect+to+a+project).

## Terminology

??? info "Understand the difference between _configuration_ and _experiment_"
    {% include-markdown "includes/terminology.md"
        start="<!--start:terminology-conf-vs-exp-->"
        end="<!--start:terminology-conf-vs-exp-->"
    %}

## _Rose/Cylc_ workflow manager

{% include-markdown "includes/rose_cylc8.md"
   start="<!--start:cylc8-compatibility-mode-->"
   end="<!--end:cylc8-compatibility-mode-->"
%}

{% include-markdown "includes/rose_cylc8.md"
   start="<!--start:cylc8-about-->"
   end="<!--end:cylc8-about-->"
%}

??? info "_Rose/Cylc_ directory and files organisation" 
 
    {% include-markdown "includes/rose_cylc8.md"
        start="<!--start:cylc8-structure-->"
        end="<!--end:cylc8-structure-->"
    %}

## Connect to _Gadi_

{% include-markdown "includes/rose_cylc8.md"
    start="<!--start:cylc8-gadi-->"
    end="<!--end:cylc8-gadi-->"
%}

## Setup your _GitHub_ account

> To work with {{model}}, you need to ensure you can identify your _GitHub_ account from _Gadi_. If you do nothing, you can simply enter your _GitHub_ credentials for all the interactions with _GitHub_ from _Gadi_. It is however possible to setup the exchange of credentials between _Gadi_ and _GitHub_ to avoid entering them every time. <br>

Reviewer 2: ```suggestion
To work with {{model}}, you need to authenticate your _GitHub_ account from _Gadi_. You can enter your _GitHub_ credentials each time you interact with _GitHub_, or configure credential exchange between _Gadi_ and _GitHub_ to avoid repeated authentication. <br>
```

> Several methods are available and they will all work with {{model}}. This documentation is written assuming authentication through the HTTPS protocol. If you are setup for an SSH-based authentication, you can adapt the commands in the documentation. Alternatively, you can add an HTTPS-based authentication to your account without impacting your SSH-based authentication.

Reviewer 2: ```suggestion
Several authentication methods are supported and work with {{model}}. This documentation assumes HTTPS authentication; if you use SSH authentication, adapt the commands accordingly. You can also configure HTTPS authentication alongside existing SSH authentication.
```

> We recommend using the `gh` command-line interface for the authentication between _GitHub_ and _Gadi_.

Reviewer 2: ```suggestion
We recommend using the `gh` command-line interface to authenticate _GitHub_ from _Gadi_.
```

??? info "Setup _GitHub_ authentication on _Gadi_ with `gh`"

>     1. Load the `gh` module on _Gadi_ (in a _Gadi_ terminal or a VDI terminal):

Reviewer 2: ```suggestion
    1. Load the `gh` module on _Gadi_ from a _Gadi_ or VDI terminal:
```

        ```
        module use /g/data/vk83/modules
        module load gh/2.49.2-1
        ```
    

>     2. Run `gh auth login` and follow the prompts to use the HTTPS protocol. Note that there is no browser installed on _Gadi_, you will need to copy/paste the given URL to your own browser window.

Reviewer 2: ```suggestion
    2. Run `gh auth login` and follow the prompts to authenticate using HTTPS. As _Gadi_ does not have a browser, copy and paste the provided URL into your local browser.
```

            <terminal-line data="input">gh auth login</terminal-line>
            <terminal-line>? What account do you want to log into? <span class="spack-cyan">GitHub.com</span></terminal-line>
            <terminal-line>? What is your preferred protocol for Git operations on this host? <span class="spack-cyan">HTTPS</span></terminal-line>
            <terminal-line>? How would you like to authenticate GitHub CLI? <span class="spack-cyan">Login with a web browser</span></terminal-line>
            <terminal-line></terminal-line>
            <terminal-line>! First copy your one-time code: <span class="payu-dark-yellow">XXXX-XXXX</span></terminal-line>
            <terminal-line>Press Enter to open github.com in your browser...</terminal-line>
            <terminal-line>! Failed opening a web browser at https://github.com/login/device</terminal-line>
            <terminal-line>exec: "xdg-open,x-www-browser,www-browser,wslview": executable file not found in $PATH</terminal-line>
            <terminal-line>Please try entering the URL in your browser manually</terminal-line>
        </terminal-window>

??? info "Email verification in _GitHub_ for source code modifications"

>     If you are planning to modify the source code of {{model}}, you will need to have your institutional email attached to your GitHub account and have the email verified. Please follow [GitHub's instructions](https://docs.github.com/en/account-and-profile/concepts/email-addresses) for that process.

Reviewer 2: ```suggestion
If you plan to modify the {{model}} source code, you must have your institutional email address linked to and verified on your _GitHub_ account. Follow [_GitHub_’s instructions](https://docs.github.com/en/account-and-profile/concepts/email-addresses) to set this up. 
```

> 

Author: This is not a requirement at this stage. It will become one when the code repos are forks of UKMO's or when the UK licence account will be GitHub-based and not MOSRS-based. 

I figured it didn't hurt to add the instructions anyway. 

It's not necessary for a beginner How to run guide, but it's a setup step so seems well placed here instead of popping out as a surprise in the config docs where instructions on changing the code and building the code might be.

    {% include-markdown "includes/persistent-sessions.md"
        start="<!--start:pers-session-start-->"
        end="<!--end:pers-session-start-->"
    %}

??? info "Assign the persistent session to _Cylc_ (once only)"

    {% include-markdown "includes/persistent-sessions.md"
        start="<!--start:pers-session-assign-->"
        end="<!--end:pers-session-assign-->"
    %}

??? info "Setup the connection between _Cylc_ and _Gadi_ (once only)"

    {% include-markdown "includes/persistent-sessions.md"
        start="<!--start:pers-session-setup-->"
        end="<!--end:pers-session-setup-->"
    %}

??? info "List active persistent sessions"

    {% include-markdown "includes/persistent-sessions.md"
        start="<!--start:pers-session-active-->"
        end="<!--end:pers-session-active-->"
    %}

??? info "Terminate a persistent session"

    {% include-markdown "includes/persistent-sessions.md"
        start="<!--start:pers-session-terminate-->"
        end="<!--end:pers-session-terminate-->"
    %}

## Access _Rose/Cylc_

{% include-markdown "includes/rose_cylc8.md"
    start="<!--start:cylc8-module-->"
    end="<!--end:cylc8-module-->"
%}

## Get {{model}} configuration

> All released {{ model }} configurations are available from the [{{ model }} configs]({{github_configs}}) GitHub repository: `{{github_configs}}`.<br>

Reviewer 2: ```suggestion
All released {{ model }} configurations are available in the [{{ model }} configs]({{github_configs}}) _GitHub_ repository: `{{github_configs}}` <br>
```

Supported configurations:

| Configuration                             | Branch name            |
|-------------------------------------------|------------------------|
| Low resolution based on GC9 configuration | release-n96e           |
| High resolution with climatology aerosols | release-n512e-aeroclim |

{% include-markdown "includes/rose_cylc.md" 
    start="<!--start:get-github-config-->" 
    end="<!--end:get-github-config-->" 
%}

## Initial Configuration Setup

> The configuration is setup to use your default project for data storage and compute resources. To choose a different project, in the _configuration directory_, open the `rose-suite.conf` file and change:

Reviewer 2: ```suggestion
The configuration is set up to use your default project for data storage and compute resources. To use a different project, open `rose-suite.conf` in the _configuration directory_ and change:
```

> - `STORAGE_PROJECT` &rarr; Specify the project to use for storage. Experiment files will be stored under `/scratch/<STORAGE_PROJECT>/$USER/cylc-run`.

Reviewer 2: ```suggestion
- `STORAGE_PROJECT`- specifies the project used for storage. Experiment files are stored under `/scratch/<STORAGE_PROJECT>/$USER/cylc-run`. If you change STORAGE_PROJECT from your default project, note the [additional installation step](https://access-hive-docs--1238.org.readthedocs.build/models/run_a_model/run_access-am3/#run-the-experiment) described below.
```

> - `COMPUTE_PROJECT` &rarr; Specify the project you want to be charged for the compute resources.

Reviewer 2: ```suggestion
- `COMPUTE_PROJECT` - specifies the project charged for the compute resources.
```
